models/svc/vits/vits.py

```diff
@@ -240,24 +238,23 @@ def forward(self, data):
     @torch.no_grad()
     def infer(self, data, noise_scale=0.35, seed=52468):
         # c, f0, uv, g
-        c = data["contentvec_feat"].transpose(1, 2)
         f0 = data["frame_pitch"]
         g = data["spk_id"]
 
-        if c.device == torch.device("cuda"):
+        if f0.device == torch.device("cuda"):
```

Comment:
How about if the input features don't contain F0? If you need to get the frame length, use the `target_len`. I'm not sure about how to get the specific device, maybe you can follow [this line](https://github.com/open-mmlab/Amphion/blob/4e8e7a5a5975420413b1f6fd338757314d882741/models/svc/diffusion/diffusion_trainer.py#L65C41-L65C41).

Reply:
Thanks! I'll detach the data dependency from f0 in the future. While f0 is always a  necessity since default generator needs it(nsf-hifigan).